File: NapakalakiRuby1/lib/cultist_player.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'cultist.rb'
require_relative 'player.rb'

module NapakalakiGame
  class CultistPlayer < Player
    @@totalCultistPlayers = 0

    def initialize (p, c)
      super(p.name)
      super.copia(p)
      @myCultistCard=c
      @@totalCultistPlayer+=1
    end
    def self.totalCultistPlayers
      return @@totalCultistPlayers
    end
    def getCombatLevel
      uno = super.getCombatLevel
      dos = super.getCombatLevel/5
      tres = @myCultistCard.getGainedLevels
      lvl = (uno+dos+tres)*@@totalCultistPlayers
      return lvl
    end
    def getOponentLevel (m)
      return m.getCombatLevelAgainstCultistPlayer
    end
    def shouldConvert
      return false
    end
    def to_s
      super + ", #{@myCultistCard.name}"
    end
    protected :getCombatLevel, :getOponentLevel, :shouldConvert
  end
end

File: Napakalaki1/src/napakalaki1/CultistPlayer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki1;

/**
 *
 * @author TOSHIBA
 */
public class CultistPlayer extends Player{
    private static int totalCultistPlayers =0;
    Cultist myCultistCard;
    
    public CultistPlayer(Player p, Cultist c){
        super(p);
        myCultistCard = c;
        totalCultistPlayers++;
    }
    @Override
    public int getCombatLevel(){
       int uno = super.getCombatLevel();
       int dos = (super.getCombatLevel()/5);
       int tres = myCultistCard.getGainedLevels();
       int lvl = (uno + dos + tres)*totalCultistPlayers;
       super.cLSectario=lvl;
       return lvl;
    }
    @Override
    public int getOponentLevel( Monster m){
        return m.getCombatLevelAgainstCultistPlayer();
    }
    @Override
    public boolean shouldConvert(){
        return false;
    }
    private Treasure giveMeATreasure(){
        int random = (int) (Math.random()*super.getVisibleTreasures().size());
        return super.getVisibleTreasures().get(random);
    }
    
    private boolean canYouGiveMeATreasure(){
        if(enemy.getVisibleTreasures().size()>0){
            return true;
        }
        else{
            return false;
        }
    }
    public int getTotalCultistPlayers(){
        return totalCultistPlayers;
    }
}
